src/ompl/geometric/planners/informedtrees/bitstar/src/ImplicitGraph.cpp

```diff
@@ -1093,8 +1096,14 @@ namespace ompl
                 // Run until at the end:
                 while (goalIter != goalEnd)
                 {
-                    // Check if this goal has met the criteria to be pruned
-                    if (this->canSampleBePruned(*goalIter))
+                    // Check if this goal has met the criteria to be pruned, but make sure it is not the goal that is
+                    // the current best solution. The current goal can satisfy the condition for pruning if the
+                    // heuristic is perfect, since samples are pruned unless the combined costs of the heuristic
+                    // cost-to-come and heuristic cost-to-go is (strictly) lower than the current solution cost. If the
+                    // heuristic is perfect, then the heuristic cost-to-come is equal to the solution cost (and the
+                    // heuristic cost-to-go is zero, since this is a goal).
+                    if (this->canSampleBePruned(*goalIter) &&
+                        costHelpPtr_->isCostNotEquivalentTo((*goalIter)->getCost(), solutionCost_))
```

Comment:
Expanding the comparison inequality for goals from `<` to `<=` seems like a harmless and way to check if the goal-considered-for-removale is part of the current solution. It's slightly indirect, but I don't know if there's a direct way to check if a vertex is part of the current solution. And I don't see any harm in the rare (impossible?) case of keeping a goal that meets equality but is not part of the current solution.